src/sentry/static/sentry/app/views/eventsV2/results.tsx

```diff
@@ -99,9 +106,70 @@ class Results extends React.Component<Props, State> {
     }
   }
 
-  async fetchTotalCount() {
+  canLoadEvents = async () => {
+    const {api, location, organization} = this.props;
+    const {eventView} = this.state;
+    let needConfirmation = false;
+    let confirmedQuery = true;
+    const currentQuery = eventView.getEventsAPIPayload(location);
+    const duration = eventView.getDays();
+
+    if (duration > 30 && currentQuery.project) {
+      let projectLength = currentQuery.project.length;
+
+      if (
+        projectLength === 0 ||
+        (projectLength === 1 && currentQuery.project[0] === '-1')
+      ) {
+        const results = await fetchProjectsCount(api, organization.slug);
+
+        if (projectLength === 0) projectLength = results.myProjects;
+        else projectLength = results.allProjects;
+      }
+
+      if (projectLength > 10) {
+        needConfirmation = true;
+        confirmedQuery = false;
+      }
+    }
+    // Once confirmed, a change of project or datetime will happen before this can set it to false,
+    // this means a query will still happen even if the new conditions need confirmation
+    // using a state callback to return this to false
+    this.setState({needConfirmation, confirmedQuery}, () => {
+      this.setState({confirmedQuery: false});
+    });
+    if (needConfirmation) {
+      this.openConfirm();
```

Comment:
This should be fine so long as `<Confirm>` is rendered and calls `setOpenFunction` before this function, just double check that `openConfirm` is set when the component calls `componentDidMount` as that should be the first lifecycle hook which calls this function, as far as my rudimentary React knowledge goes.

Reply:
:+1: confirmed, added a debugger in `componentDidMount` and checked that `openConfirm` was already set